getting-started/authenticating-users/app.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The main program is a sample web server application that extracts and
+// verifies user identity data passed to it via Identity-Aware Proxy.
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+var cachedAudience string
+var cachedCertificates = make(map[string]string)
```

Comment:
Will updates to these will cause race conditions? I'm not sure a good way to cache without adding locking, which makes the sample more complex. But, more complex & no race is probably better than simple & race. Maybe don't cache?

It's also generally not recommended to have global variables. A common practice is to create an `app` struct that holds the variables you need, like this: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/getting-started/gopher-run/cmd/main.go#L40.

Reply:
No, I don't see that they would, since the assignment would always set the same value. However, since I'm now using the metadata library, I'll let it worry about caching for optimization and remove this cache.